### DISCUSSION TITLE
TypeVarTuple and tuple[Any, ...]

### DISCUSSION THREAD

**Discussion by mehdigmira** [microsoft/pyright#5861] [General]
Hello,

I'm not sure this behavior is expected

```python
from typing import TypeVarTuple, Generic, Any

Ts = TypeVarTuple("Ts")


class Array(Generic[*Ts]):
    ...


def process(x: Array[*tuple[Any, ...]]) -> None:
    ...


def f(x: Array[*Ts]):
    process(x)  # Error # "Array[*Ts@f]" is incompatible with "Array[*tuple[Any, ...]]"


def g(x: Array[str, int]):
    process(x)  # No error
```

Array[*Ts] is considered as not compatible with Array[*tuple[Any, ...]], but Array[str, int] is valid.

**Comment by erictraut** [comment]
For consistency, `*tuple[Any, ...]` should be assignable in this case. That type is the moral equivalent of `Any` for a TypeVarTuple, just as `...` is the moral equivalent of `Any` for a ParamSpec.
